analytics_data_api/utils.py

```diff
@@ -31,3 +31,53 @@ def set_user_auth_token(user, key):
     Token.objects.create(user=user, key=key)
 
     print "Set API key for user %s to %s" % (user, key)
+
+
+def matches(answer_one, answer_two):
+    """ Check whether two answers match sufficiently for consolidation. """
+    return (
+        answer_one.question_text == answer_two.question_text and
+        answer_one.answer_value_text == answer_two.answer_value_text and
+        answer_one.answer_value_numeric == answer_two.answer_value_numeric and
+        answer_one.problem_display_name == answer_two.problem_display_name and
+        answer_one.correct == answer_two.correct and
+        answer_one.value_id == answer_two.value_id
+    )
+
+
+def consolidate_answers(problem):
+    """ Attempt to consolidate erroneously randomized answers. """
+    value_id_sets = {}
+
+    for answer in problem:
+        answer.consolidated_variant = False
+
+        if answer.value_id in value_id_sets:
+            value_id_sets[answer.value_id].append(answer)
+        else:
+            value_id_sets[answer.value_id] = [answer]
+
+    consolidated_answers = []
+
+    for _, answers in sorted(value_id_sets.iteritems(), reverse=True):
+        consolidate = len(answers) > 1
+
+        for i, answer in enumerate(answers):
+            j = i + 1
+
+            while consolidate and j < len(answers):
+                if not matches(answers[i], answers[j]):
+                    consolidate = False
+                else:
+                    j += 1
```

Comment:
just recapping a hipchat discussion, I think this logic can be simplified by using python's `set` on tuples that are generated from `answers` like how `matches` does it now.